README.md

```diff
@@ -79,68 +83,161 @@ following fixtures:
 
 import pytest
 from ragger import Firmware
-from ragger.backend import SpeculosBackend, LedgerCommBackend
+from ragger.backend import SpeculosBackend, LedgerCommBackend, LedgerWalletBackend
+from ragger.navigator import NanoNavigator
+from ragger.utils import app_path_from_app_name
+
 
 # This variable is needed for Speculos only (physical tests need the application to be already installed)
-APPLICATION = "Path/to/the/application.elf"
-# This variable will be useful in tests to implement different behavior depending on the firmware
-NANOS_FIRMWARE = Firmware("nanos", "2.1")
+APPS_DIRECTORY = (Path(__file__).parent.parent / "elfs").resolve()
+
+APP_NAME = "MyAPP"
+
+BACKENDS = ["speculos", "ledgercomm", "ledgerwallet"]
+
+FIRMWARES = [Firmware('nanos', '2.1'),
+             Firmware('nanox', '2.0.2'),
+             Firmware('nanosp', '1.0.3')]
+
 
-# adding a pytest CLI option "--backend"
 def pytest_addoption(parser):
-    print(help(parser.addoption))
     parser.addoption("--backend", action="store", default="speculos")
+    parser.addoption("--display", action="store_true", default=False)
+    # Enable using --'device' in the pytest command line to restrict testing to specific devices
+    for fw in FIRMWARES:
+        parser.addoption("--"+fw.device, action="store_true", help="run on nanos only")
+
 
-# accessing the value of the "--backend" option as a fixture
 @pytest.fixture(scope="session")
 def backend(pytestconfig):
     return pytestconfig.getoption("backend")
 
-# Providing the firmware as a fixture
-# This can be easily parametrized, which would allow to run the tests on several firmware type or version
+
+@pytest.fixture(scope="session")
+def display(pytestconfig):
+    return pytestconfig.getoption("display")
+
+
 @pytest.fixture
-def firmware():
-    return NANOS_FIRMWARE
+def test_name(request):
```

Comment:
Maybe make it a property ?

Even make it two properties:
self.test_name => return request.node.name
self.test_name_no_suffix => return self.test_name.split("[")[0]